crates/containerd-shim-wasmtime/src/instance.rs

```diff
@@ -1,32 +1,49 @@
+use std::collections::hash_map::DefaultHasher;
 use std::fs::File;
-use std::sync::OnceLock;
+use std::hash::{Hash, Hasher};
+use std::marker::PhantomData;
 
 use anyhow::{bail, Context, Result};
 use containerd_shim_wasm::container::{
-    Engine, Entrypoint, Instance, RuntimeContext, RuntimeInfo, Stdio, WasmBinaryType,
+    Engine, Entrypoint, Instance, RuntimeContext, Stdio, WasmBinaryType,
 };
 use wasi_common::I32Exit;
 use wasmtime::component::{self as wasmtime_component, Component, ResourceTable};
-use wasmtime::{Module, Precompiled, Store};
+use wasmtime::{Config, Module, Precompiled, Store};
 use wasmtime_wasi::preview2::{self as wasi_preview2};
 use wasmtime_wasi::{self as wasi_preview1, Dir};
 
-pub type WasmtimeInstance = Instance<WasmtimeEngine>;
+pub type WasmtimeInstance = Instance<WasmtimeEngine<DefaultConfig>>;
 
 #[derive(Clone)]
-pub struct WasmtimeEngine {
+pub struct WasmtimeEngine<T: WasiConfig + Clone + Send + Sync> {
     engine: wasmtime::Engine,
+    config_type: PhantomData<T>,
 }
 
-impl Default for WasmtimeEngine {
-    fn default() -> Self {
+#[derive(Clone)]
+pub struct DefaultConfig {}
+
+impl WasiConfig for DefaultConfig {
+    fn new_config() -> Config {
         let mut config = wasmtime::Config::new();
-        config.parallel_compilation(false);
         config.wasm_component_model(true); // enable component linking
+        config
+    }
+}
+
+pub trait WasiConfig {
```

Comment:
WasiConfig could include `Clone + Sync + Send` as supertraits

```rust
pub trait WasiConfig: Clone + Sync + Send {
    fn new_config() -> Config;
}
```

Reply:
ok, I thought I tried this but maybe go the syntax wrong, will try again